### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by IlyaGrynyshyn**

### CODE REVIEW

File: cinema/views.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+@api_view(["GET", "POST"])
+def get_all_movies(request):
+    if request.method == "GET":
+        movies = Movie.objects.all()
+        serializer = MovieSerializer(movies, many=True)
+        return Response(serializer.data, status=200)
+    if request.method == 'POST':
+        serializer = MovieSerializer(data=request.data)
+        if serializer.is_valid(raise_exception=True):
```
Comment (vasylhnatiuk) [inline cinema/views.py:17]:
```suggestion
        serializer.is_valid(raise_exception=True):
```

File: cinema/views.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+            serializer.save()
+            return Response(serializer.data, status=201)
+        return Response(serializer.errors, status=400)
```
Comment (vasylhnatiuk) [inline cinema/views.py:20]: serializer.is_valid(raise_exception=True) will do it automatically. delete this line

File: cinema/views.py [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+@api_view(["PUT", "DELETE", "GET"])
+def movie_detail(request, pk):
+    movie = get_object_or_404(Movie, pk=pk)
+    if request.method == "GET":
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=200)
+
+    if request.method == 'PUT':
+        serializer = MovieSerializer(movie, data=request.data)
+        if serializer.is_valid():
```
Comment (vasylhnatiuk) [inline cinema/views.py:32]: `if `useless here

**Review by vasylhnatiuk — CHANGES_REQUESTED** [review]

File: cinema/views.py [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+
+    if request.method == "PUT":
+        serializer = MovieSerializer(movie, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=201)
+
+    if request.method == "DELETE":
+        movie.delete()
+        return Response(status=404)
```
Comment (pavlejviki) [inline cinema/views.py:37]: Status code 404 (Not Found) is generally used when a resource cannot be found. For a successful DELETE operation, the typical status code is 204 (No Content), indicating that the action was successful and there is no content to return.

File: cinema/views.py [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    movie = get_object_or_404(Movie, pk=pk)
+    if request.method == "GET":
+        serializer = MovieSerializer(movie)
+        return Response(serializer.data, status=200)
+
+    if request.method == "PUT":
+        serializer = MovieSerializer(movie, data=request.data)
+        serializer.is_valid(raise_exception=True)
+        serializer.save()
+        return Response(serializer.data, status=201)
```
Comment (pavlejviki) [inline cinema/views.py:33]: since it's an update to an existing resource, 200 (OK) is more suitable

**Review by pavlejviki — CHANGES_REQUESTED** [review]

**Review by Polyakiv-Andrey — APPROVED** [review]
Good job!

**Pull request closed without merging**